src/object-pool.ts

```diff
@@ -1,10 +1,287 @@
-import {ArmorObjectPoolOptions} from './object-pool-options';
-import {ArmorObjectPoolState} from './object-pool-state';
+import ArmorCollection from './collection';
+import ArmorCollectionSelector from './selector';
+import ArmorObjectPoolInstance from './object-pool-instance';
+import ArmorObjectPoolOptions from './object-pool-options';
+import ArmorObjectPoolState from './object-pool-state';
 
-export class ArmorObjectPool<T> {
-	public readonly state: ArmorObjectPoolState;
+export default class ArmorObjectPool<T> implements ArmorCollection<T> {
+	public state: ArmorObjectPoolState<T>;
+	public startSize: number = 10;
+	public readonly poolObj: ArmorObjectPoolInstance<T>;
 
-	constructor(options?: ArmorObjectPoolOptions) {
-		this.state = new ArmorObjectPoolState(options);
+	constructor(poolObj: ArmorObjectPoolInstance<T>, options?: ArmorObjectPoolOptions<T>) {
+		if (typeof poolObj !== 'function') {
```

Comment:
Assuming this message was copied from the PQ ADT. Update the message for this ADT.
Additionally, can you clarify through naming what poolObj is or does here?